packages/switch/src/switch.ts

```diff
@@ -15,31 +15,133 @@ import {
+                    value(event);
+                }
+            };
+        } else if (typeof value === 'boolean') {
+            delete inputProps[key];
+            inputProps[`?${key}`] = value;
+        }
+    }
+}
+
+const defer = Promise.resolve().then.bind(Promise.resolve());
+
+export class Switch extends Focusable implements CheckboxBase {
+    @property({ attribute: false })
+    public defaultSelected?: boolean;
+
+    @property({ type: Boolean, reflect: true, attribute: 'checked' })
+    public isSelected?: boolean;
+
+    // react-stately will expect 'on' handlers for each state change
+    // this is where we apply changes to local state, and dispatch events
+    public onChange = (isSelected: boolean): void => {
+        const changeEvent = new CustomEvent('change', {
+            bubbles: true,
+            cancelable: true,
+        });
+        const apply = this.dispatchEvent(changeEvent);
+        if (apply) {
+            this.isSelected = isSelected;
+        }
+    };
+
+    @property({ reflect: true })
+    public value?: string;
+
+    @property({ reflect: true })
+    public name?: string;
+
+    // haunted specific: changes to lit-element, could be included in base class
+    private hauntedState = new State(() => this.requestUpdate(), this);
+
+    protected update(changedProperties: Map<string, boolean>): void {
+        this.hauntedState.run(() => super.update(changedProperties));
+    }
+
+    protected updated(changedProperties: PropertyValues): void {
+        super.updated(changedProperties);
+        this.hauntedState.runLayoutEffects();
+        defer(() => this.hauntedState.runEffects());
+    }
+
+    public disconnectedCallback(): void {
+        this.hauntedState.teardown();
+        super.disconnectedCallback();
+    }
+    // end haunted specific
+
+    // had to bring this up from checkbox base since we're not basing off that class here
+    @query('#input')
+    protected inputElement!: HTMLInputElement;
+
+    public get focusElement(): HTMLElement {
+        return this.inputElement;
+    }
+
     public static get styles(): CSSResultArray {
         /* istanbul ignore if */
         if (window.hasOwnProperty('ShadyDOM')) {
             // Override some styles if we are using the web component polyfill
-            return [...super.styles, switchStyles, legacyStyles];
+            return [switchStyles, legacyStyles];
         }
-        return [...super.styles, switchStyles];
+        return [switchStyles];
     }
 
     protected render(): TemplateResult {
+        // use the state from react-stately, this manages isSelected state
+        const state = useToggleState(this);
+        // use the hook from react-aria, this gives us event bindings and aria
+        // attributes to use in our DOM rendering
+        const { inputProps } = useSwitch(this, state, this);
+        // we have to hack the spread naming to be compatible with lit-html
+        fixSpreadNaming(inputProps);
```

Comment:
I was definitely in conversation about whether it expand support for spreading events or booleans only, which would make this a filter rather than a conversion step 🙈 We could certainly expend the `spread` directive to support that on our own. Not sure how much we'd benefit from not having to `delete` those properties on every render...

Reply:
Agreed I figure we could build a spread helper thats suitable for this specific operation easily enough.